nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SampleRecord.java

```diff
@@ -0,0 +1,400 @@
+        if (context.isExpressionLanguageSupported(subject) && context.isExpressionLanguagePresent(value)) {
+            return new ValidationResult.Builder().subject(subject).input(value).explanation("Expression Language Present").valid(true).build();
+        }
+
+        String reason = null;
+        int percentValue;
+        try {
+            percentValue = Integer.parseInt(value);
+            if (percentValue < 0 || percentValue > 100) {
+                reason = "not a percent value (0-100)";
+            }
+        } catch (final NumberFormatException e) {
+            reason = "not a valid integer";
+        }
+
+        return new ValidationResult.Builder().subject(subject).input(value).explanation(reason).valid(reason == null).build();
+    };
+
+    static final PropertyDescriptor RECORD_READER_FACTORY = new PropertyDescriptor.Builder()
+            .name("record-reader")
+            .displayName("Record Reader")
+            .description("Specifies the Controller Service to use for parsing incoming data and determining the data's schema")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .build();
+    static final PropertyDescriptor RECORD_WRITER_FACTORY = new PropertyDescriptor.Builder()
+            .name("record-writer")
+            .displayName("Record Writer")
+            .description("Specifies the Controller Service to use for writing results to a FlowFile")
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .required(true)
+            .build();
+    static final PropertyDescriptor SAMPLING_STRATEGY = new PropertyDescriptor.Builder()
+            .name("sample-record-sampling-strategy")
+            .displayName("Sampling Strategy")
+            .description("Specifies which method to use for sampling records from the incoming FlowFile")
+            .allowableValues(INTERVAL_SAMPLING, PROBABILISTIC_SAMPLING, RESERVOIR_SAMPLING)
+            .required(true)
+            .defaultValue(RESERVOIR_SAMPLING.getValue())
+            .addValidator(Validator.VALID)
+            .build();
+    static final PropertyDescriptor SAMPLING_INTERVAL = new PropertyDescriptor.Builder()
+            .name("sample-record-interval")
+            .displayName("Sampling Interval")
+            .description("Specifies the number of records to skip before writing a record to the outgoing FlowFile. This property is only "
+                    + "used if Sampling Strategy is set to Interval Sampling. A value of zero (0) will cause all records to be included in the"
+                    + "outgoing FlowFile.")
+            .required(false)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+    static final PropertyDescriptor SAMPLING_PROBABILITY = new PropertyDescriptor.Builder()
```

Comment:
Would it make sense to add another property setting the random seed?

Reply:
Yes (most likely for testing though), will add the property and documentation